Controllers/ValuesController.cs

```diff
@@ -3,99 +3,78 @@
 
 namespace incrementally.Controllers
 {
-  [Route("api")]
-  [ApiController]
-  [Authorize]
-  public class ValuesController : ControllerBase
-  {
-    private readonly CosmosDbService _cosmosDbService;
-    private readonly PostgresCompiler _compiler;
-    public ValuesController(CosmosDbService cosmosDbService, PostgresCompiler compiler)
+    [Route("api")]
+    [ApiController]
+    [Authorize]
+    public class ValuesController : ControllerBase
     {
-      _cosmosDbService = cosmosDbService;
-      _compiler = compiler;
-    }
+        private readonly RecordingHandler _recordingHandler;
 
-    [HttpGet]
-    [AllowAnonymous]
-    public string Get()
-    {
-      return "Server is running";
-    }
+        public ValuesController(RecordingHandler recordingHandler)
+        {
+            _recordingHandler = recordingHandler;
+        }
 
-    [HttpGet]
-    [AllowAnonymous]
-    [Route("recording/{id?}")]
-    public async Task<IEnumerable<RecordingEntry>> Recording(string id)
-    {
-      return await _cosmosDbService.GetRecordings(id);
-    }
+        [HttpGet]
+        [AllowAnonymous]
+        public string Get()
+        {
+            return "Server is running";
+        }
 
-    [HttpGet]
-    [AllowAnonymous]
-    [Route("metadata/{id}")]
-    public async Task<IEnumerable<RecordingMetadata>> SingleMetadata(string id)
-    {
-      return await _cosmosDbService.GetRecordingMetadata(id);
-    }
+        [HttpGet]
+        [AllowAnonymous]
+        [Route("recording/{id?}")]
+        public async Task<RecordingEntry> Recording(string id)
+        {
+            return await _recordingHandler.GetRecording(id);
+        }
 
-    [HttpGet]
-    [AllowAnonymous]
-    [Route("metadata")]
-    public async Task<IEnumerable<RecordingMetadata>> Metadata()
-    {
-      return await _cosmosDbService.GetTopRecordingMetadata();
-    }
+        [HttpGet]
+        [AllowAnonymous]
+        [Route("metadata/{id}")]
+        public async Task<RecordingMetadata> SingleMetadata(string id)
+        {
+            return await _recordingHandler.GetMetadata(id);
+        }
 
-    [HttpGet]
-    [Route("allRecordingsMetadata")]
-    public async Task<IEnumerable<RecordingEntry>> getAll()
-    {
-      return await _cosmosDbService.GetItemsAsync($@" 
-        SELECT 
-          c.title,
-          c.description,
-          c.createdBy,
-          c.givenName,
-          c.surname,
-          c.id
-        FROM c
-        WHERE c.CreatedBy = ""{User.FindFirst(ClaimTypes.NameIdentifier).Value}""
-      ");
+        [HttpGet]
+        [AllowAnonymous]
+        [Route("metadata")]
+        public async Task<IEnumerable<RecordingMetadata>> Metadata()
+        {
+            return await _recordingHandler.GetTopMetadata();
+        }
+
+
+        [HttpPost]
+        [Route("create")]
+        public async Task<RecordingMetadata> CreateAsync(UserRecordingInput data)
+        {
+            var recordingMetadata = new RecordingMetadata();

```

Comment:
Seeing these lines I'm getting a feeling that a builder-pattern would be suitable for readability, but again I'm quite sure if this applies as well in c# as it does in Java.

Reply:
Agree this is useful, but I think in this case it's probably not right pattern to use. I do see some issues with this part of the code though and I've pushed some changes that hopefully will make it look a little nicer.